test/puppeteer_request_interception.test.js

```diff
@@ -155,6 +157,56 @@ describe('Apify.utils.puppeteer.addInterceptRequestHandler|removeInterceptReques
             await browser.close();
         }
     });
+
+    describe('internal handleRequest function should return correctly formated headers', () => {
+        const HOSTNAME = '127.0.0.1';
+        let port;
+        let server;
+        beforeAll(async () => {
+            const app = express();
+
+            app.get('/getRawHeaders', (req, res) => {
+                res.send(JSON.stringify(req.rawHeaders));
+            });
+
+            server = await startExpressAppPromise(app, 0);
+            port = server.address().port; //eslint-disable-line
+        });
+
+        afterAll(() => {
+            server.close();
+        });
+
+        test('should correctly capitalize headers', async () => {
+            const browser = await Apify.launchPuppeteer({ headless: true });
+
+            try {
+                const page = await browser.newPage();
+
+                await addInterceptRequestHandler(page, async (request) => {
+                    // Override headers
+                    const headers = {
+                        ...request.headers(),
+                        accept: 'text/hml',
+                        'accept-language': 'en-GB',
+                        'upgrade-insecure-requests': 2,
+                    };
+                    return request.continue({ headers });
+                });
+
+                const response = await page.goto(`http://${HOSTNAME}:${port}/getRawHeaders`);
+                const rawHeadersArr = JSON.parse(await response.text());
+
+                expect(rawHeadersArr.includes('Accept')).toEqual(true);
+                expect(rawHeadersArr.includes('Accept-Language')).toEqual(true);
+                expect(rawHeadersArr.includes('Upgrade-Insecure-Requests')).toEqual(true);
+
+                expect(rawHeadersArr.includes('Connection')).toEqual(true); // default should be capitalized too
```

Comment:
You should check the values to make sure they were actually overridden.

Reply:
Ok